staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -234,6 +234,23 @@ type Store struct {
 	// If set, DestroyFunc has to be implemented in thread-safe way and
 	// be prepared for being called more than once.
 	DestroyFunc func()
+
+	// CorruptObjDeleter, if initialized, will make an attempt to
+	// perform the normal deletion flow, but if either of the below occurs:
+	// a) the data (associated with the resource being deleted) retrieved
+	// from the storage failed to transform properly (eg. decryption failure)
+	// b) the data (associated with the resource being deleted) failed to
+	// decode properly (eg. corrupt data)
+	// it will disregard these errors, bypass the finalzer constraints,
+	// deletion hook(s) and go ahead with the deletion of the object.
+	//
+	// WARNING: This may break the cluster if the resource has
+	// dependencies. Use when the cluster is broken, and there is no
+	// other viable option to repair the cluster.
```

Comment:
```suggestion
	// WARNING: This may break the cluster e.g. if the resource has
	// dependencies. Use when the cluster is broken, and there is no
	// other viable option to repair the cluster.
```